### DISCUSSION TITLE
Fonts and sizes changing by themselves [SOLVED]

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2371] [Q&A] [answered]
I noticed yesterday when I launched QTerminal, the font in the screen had unexpectedly changed by itself. It was on Monospace 11. I looked at the other fonts listed (Debian unstable/sid) and changed it to DejaVu Sans Mono 13, which is what the font looked like previously.

This morning, I discovered the font under the icons on the desktop were smaller. The default showed Sans 12, but there is no "Sans" font installed (at least with just that name). I changed this font to Sans Serif 12, which was the closest to the previous font before it changed.

The display is a Samsung 22-inch LCD, connected to the PC via a DVI-to-HDMI cable and the resolution is 1920x1080.

Is there something causing these fonts to change by themselves all of a sudden?

Thanks in advance.

**Comment by tsujan** [comment]
> The display is ... and the resolution is 1920x1080.

The monitor or its resolution can't change the font. I've used LXQt with 4 1920x1080 screens, LCD and LED, but have never seen a spontaneous font change, not even with multi-screen setups.

Your problem should have another cause, but I'm afraid I have no more idea.

**Reply by edwardp3** [reply]
@tsujan Thank you for the reply. I included the monitor/display information, in case it was related somehow.

Numerous Qt-related packages had upgrades in Debian sid/unstable. Before I upgraded my other desktop, I made note of the font settings. After the upgrade, some of the fonts, did in fact, change...

**Desktop Preferences/Label Text**: Font remained "Sans" 11, but on the desktop the font under the icons was noticeably smaller in size, plus the spacing between the icons became wider and I was unable to change this spacing initially. The icons snapped back to the initial space on the desktop, until that particular font was changed. Font was changed to DejaVu Sans 12 Book, the font became larger, the spacing between the icons narrowed and I was then able to move them back to their original positions on the desktop.

**LXQt Appearance Configuration/Font**: Changed to Noto Sans 11 (was originally DejaVu Sans 11) and was changed back to DejaVu Sans 11.

**Openbox Settings/Font**: Active and Inactive title fonts changed to "sans" (lower case) 8 (was originally size 11). These were changed to DejaVu Sans 11 Bold.

There were no obvious font packages upgraded, so I do not know why the fonts changed after the upgrade. 

One of the packages upgraded, was specific to LXQt: **lxqt-qtplugin** (it is now on version 1.2.0.3+b2).

The corrected font settings are now the same on both desktops.

**Reply by tsujan** [reply]
I don't have Debian, but I guess your default fonts may have changed after the upgrade of some font-related packages. It happened on Arch years ago.

**Comment by M4he** [comment, marked as answer]
The default font settings on Debian are often "Sans" and "Monospace". These aren't fonts, they are [font aliases](https://wiki.archlinux.org/title/Fonts#Font_alias). Depending on which fonts you have installed, a specific font will be selected for that alias. The current matches can be checked with:

```
fc-match sans
fc-match monospace
```

If you install new fonts or packages that pull in new fonts as dependencies, the match of these aliases might change due to font packages pulling in new fontconfig files that change alias priorities and make those prefer the new fonts. Thus it might feel like your fonts randomly change.

If you configure specific fonts instead of the placeholder aliases (like you did with QTerminal), you should be safe from random changes in the future. The "Sans Serif" you selected for the desktop sounds like an alias to me as well. Better change that to "DejaVu Sans" or any font that fits.

**Reply by edwardp3** [reply]
My font selections are now actual fonts.

Sans Serif, is indeed, an alias:
```
~$ fc-match sans-serif
NotoSans-Regular.ttf: "Noto Sans" "Regular"
```

The others:

```
~$ fc-match sans
NotoSans-Regular.ttf: "Noto Sans" "Regular
```

```
~$ fc-match monospace
NotoSansMono-Regular.ttf: "Noto Sans Mono" "Regular"
```

The Noto fonts were included with the default installation.

Fonts are now set to: 

LXQT Appearance - DejaVu Sans 11 Normal

Openbox - DejaVu Sans 11 Bold (I noticed when clicking the Font name, the style in the selection window switches to Oblique as it's the first style in the list, but that's an Openbox issue.)

Qterminal: DejaVu Sans Mono 13

Desktop Preferences/Label Text - DejaVu Sans 12 Book

**Reply by M4he** [reply]
Looks good. You should be safe from any alias changes now.

**Reply by tsujan** [reply]
One or two years ago, an Arch/Manjaro upgrade, which brought Noto into my system (I hadn't installed it), changed my Monospace badly. I restored it by editing the user's fontconfig.

To set your aliases to DejaVu, you could create `~/.config/fontconfig/conf.d/0-mono.conf` as:

```xml
<?xml version='1.0'?>
<!DOCTYPE fontconfig SYSTEM 'fonts.dtd'>
<fontconfig>
 <alias>
  <family>monospace</family>
  <prefer>
   <family>DejaVu Sans Mono</family>
  </prefer>
 </alias>
 <alias>
  <family>serif</family>
  <prefer>
   <family>DejaVu Serif</family>
  </prefer>
 </alias>
 <alias>
  <family>sans-serif</family>
  <prefer>
   <family>DejaVu Sans</family>
  </prefer>
 </alias>
</fontconfig>

```

And if you have "Noto Color Emoji", you could add `~/.config/fontconfig/conf.d/0-emoji.conf` as:

```xml
<?xml version='1.0'?>
<!DOCTYPE fontconfig SYSTEM 'fonts.dtd'>
<fontconfig>
 <!-- For colored emojis: -->
 <match target="pattern">
  <test name="family">
   <string>monospace</string>
  </test>
  <edit mode="append" name="family">
   <string>Noto Color Emoji</string>
  </edit>
 </match>
 <match target="pattern">
  <test name="family">
   <string>serif</string>
  </test>
  <edit mode="append" name="family">
   <string>Noto Color Emoji</string>
  </edit>
 </match>
 <match target="pattern">
  <test name="family">
   <string>sans-serif</string>
  </test>
  <edit mode="append" name="family">
   <string>Noto Color Emoji</string>
  </edit>
 </match>
</fontconfig>

```

EDIT: The info on emojis can be found here too: https://github.com/lxqt/qterminal/wiki

**Reply by M4he** [reply]
Good point @tsujan
Some applications (especially ones you might install later on) might still use the aliases unless you configure them indivdiually. Tweaking the aliases via fontconfig makes sure that those cases are covered upfront and should also prevent any overrides/changes by system packages. So if you don't mind handling those config files, it might be an option as well.

**Reply by tsujan** [reply]
I think I've opened an issue about it — if not, it should be opened: LXQt needs a config dialog for setting the most well-known aliases. I didn't find time to do it, hoping that a contributor would.

**Comment by johndoe71rus** [comment]
Let me join the discussion. I don't think my question deserves a new thread. Lubuntu 22.04.5, installed lxqt 1.4.0 from ppa. Fonts are strange. Created ~/.config/fontconfig/conf.d/0-mono.conf and ~/.config/fontconfig/fonts.conf. Normal font looks thinner, bold font looks bolder. I have another system and compared the images.
![lxqt017_lubuntu_2204](https://github.com/user-attachments/assets/c7201ee5-d1cc-4823-9ce4-d98dbb127df8)
![lxqt_140_lubuntu_2204](https://github.com/user-attachments/assets/7c697376-212c-4714-84be-a561ebc6f7ca)
The window title is also bold, and it looks familiar.
Sorry if I should have started a separate discussion.

**Reply by tsujan** [reply]
LXQt 1.4 is outdated, let alone 0.17.

**Reply by johndoe71rus** [reply]
I understand that. For 22.04 lts lxqt 0.17 is the only option until the end. The next 24.04 lts lxqt 1.4.0. I recently learned that there is a Lubuntu Backports PPA. This is my experiment to get a familiar looking system. I try lxqt 2.0 with 24.10 and have same problem with fonts.

**Reply by tsujan** [reply]
I don't see any problem with your font. It's just different from the older one, actually better.

**Reply by johndoe71rus** [reply]
finally fixed the look. they (ubuntu team) updated the ubuntu font, it became thinner. Old style fonts-ubuntu-classic.
